### ISSUE TITLE
Panics upon pasting "<documentation>\n" on assistant panel

### ISSUE THREAD

**Issue by fdionisi** [zed-industries/zed#22236]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Copy <documentation> w/ ending new line
```
<documentation>

```
2. Open Zed
3. Cmd+v on the assistant panel
4. 💥

### Environment

Zed: v0.167.0 (Zed Preview)
OS: macOS 14.2.1
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

```
2024-12-19T07:25:40.834358-01:00 [ERROR] {
  "thread": "main",
  "payload": "offset is out of range",
  "location_data": {
    "file": "crates/multi_buffer/src/multi_buffer.rs",
    "line": 5159
  },
  "backtrace": [
    "core::panicking::panic_fmt",
    "multi_buffer::MultiBufferSnapshot::anchor_at",
    "assistant::assistant_panel::ContextEditor::paste",
    "gpui::elements::div::Interactivity::capture_action::{{closure}}",
    "gpui::window::WindowContext::dispatch_action_on_node",
    "gpui::window::WindowContext::dispatch_event",
    "gpui::window::Window::new::{{closure}}",
    "gpui::platform::mac::window::do_command_by_selector",
    "gpui::platform::mac::window::handle_key_event",
    "<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run",
    "gpui::app::App::run",
    "zed::main",
    "std::sys::backtrace::__rust_begin_short_backtrace",
    "std::rt::lang_start::{{closure}}",
    "std::rt::lang_start_internal",
    "_main"
  ],
  "app_version": "0.167.0",
  "release_channel": "Zed Preview",
  "os_name": "macOS",
  "os_version": "14.2.1",
  "architecture": "aarch64",
  "panicked_on": 1734596740834,
  "system_id": "7b8e4a4e-a372-4139-a3a8-5fd58845b995",
  "installation_id": "ea7cb936-a86f-42f2-8ad9-ea9b94ab9b88",
  "session_id": "cb278d18-9323-475b-a0ee-d46a477985bd"
}
```
</details>

**Comment by fdionisi** [comment]
I was able to consistently reproduce it (~3 times in a short amount of time, sure you'll see the installation ID), but now I cannot for some reason

**Comment by mrnugget** [comment]
I cannot reproduce this:

https://github.com/user-attachments/assets/7d646371-16e0-4a09-b619-341fa7d1086f

**Comment by fdionisi** [comment]
Yeah, I know; after changing some settings (i.e., open/close the project pane) and opening another project, it stopped panicking.

I reproduced it a few times before posting it, but something else must be enabling this behaviour.

Feel free to close the issue

**Comment by fdionisi** [comment]
@mrnugget I'm finally able to reproduce it consistently.

https://github.com/user-attachments/assets/2bea1b82-d1c6-46ff-833d-e893b145e2c1

**Comment by mrnugget** [comment]
Huh, I can't:

https://github.com/user-attachments/assets/b68db844-23bb-4141-a4e6-d73436068158

What I did:

1. Open assistant panel
2. Write `/fetch https://github.com/modelcontextprotocol/specification/blob/main/schema/schema.json`
3. Hit <kbd>return</kbd>
4. Hit <kbd>up</kbd> twice
5. Insert `<mpcspec>`
6. Hit <kbd>down</kdb> twice

**Comment by fdionisi** [comment]
Oh, I apologise reporting with the video only. I was planning to come back to it after the celebrations, but I got flu and forgot about this.

1. Open assistant panel
1. Write /fetch https://github.com/modelcontextprotocol/specification/blob/main/schema/schema.json
1. Hit return
1. Hit up twice
1. Insert <mpcspec>
1. Copy line
1. Hit down twice
1. Paste line

Yet, while I try to reproduce it on the new version it seems gone again. So, hurray for something you and the team fixed in the past days, or damn this panic is driving me crazy, I guess.

Zed: v0.168.2 (Zed Preview)
OS: macOS 14.2.1
Memory: 16 GiB
Architecture: aarch64

**Comment by bennetbo** [comment]
I'll close this one out since this is not reproducable. Also, we are in the process of completely redesigning the assistant panel, therefore it is unlikely that this also occurs in the new assistant experience
